# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/DOMTypes.ipdlh
# Commit: bb63792c3333
# Full Hash: bb63792c3333a7911bdfad9e9e9ec58bb48bad93
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 2: Create PNotification via PBackground r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227687
# ==============================================================================

diff -r 93937caf6d13 -r bb63792c3333 dom/ipc/DOMTypes.ipdlh
--- a/dom/ipc/DOMTypes.ipdlh	Thu Nov 07 17:32:03 2024 +0000
+++ b/dom/ipc/DOMTypes.ipdlh	Thu Nov 07 17:32:03 2024 +0000
@@ -12,6 +12,7 @@
 include "mozilla/dom/PropertyBagUtils.h";
 include "mozilla/dom/ReferrerInfoUtils.h";
 include "mozilla/dom/TabMessageUtils.h";
+include "mozilla/dom/notification/IPCUtils.h";
 include "mozilla/ipc/URIUtils.h";
 include "mozilla/layers/LayersMessageUtils.h";
 include "mozilla/net/NeckoMessageUtils.h";
@@ -55,6 +56,8 @@
 using mozilla::TimeStamp from "mozilla/TimeStamp.h";
 [RefCounted] using class mozilla::RemoteLazyInputStream from "mozilla/RemoteLazyInputStream.h";
 [MoveOnly] using class mozilla::ipc::BigBuffer from "mozilla/ipc/BigBuffer.h";
+using mozilla::dom::NotificationDirection from "mozilla/dom/NotificationBinding.h";
+using mozilla::dom::NotificationBehavior from "mozilla/dom/NotificationBinding.h";
 
 namespace mozilla {
 namespace dom {
@@ -190,7 +193,7 @@
   uint32_t LoadType;
   uint32_t LoadFlags;
   uint32_t InternalLoadFlags;
-  
+
   // The TriggineringSandboxFlags are the SandboxFlags of the entity
   // responsible for causing the load to occur.
   uint32_t TriggeringSandboxFlags;
@@ -329,5 +332,23 @@
     ImageIntSize size;
 };
 
+// Mostly same as NotificationOptions except:
+// * `title` is included (it's a separate parameter in the Notification constructor)
+// * `data` is serialized to base64 string by StructuredCloneContainer
+// * `vibrate` is normalized to sequence
+struct IPCNotificationOptions {
+  nsString title;
+  NotificationDirection dir;
+  nsString lang;
+  nsString body;
+  nsString tag;
+  nsString icon;
+  bool requireInteraction;
+  bool silent;
+  uint32_t[] vibrate;
+  nsString dataSerialized;
+  NotificationBehavior behavior;
+};
+
 } // namespace dom
 } // namespace mozilla